File: android/app/src/main/java/com/timaurora/tim/helper/AuthHelper.java
package com.timaurora.tim.helper;

import com.facebook.react.bridge.Promise;
import com.tencent.TIMManager;
import com.tencent.TIMUser;
import com.timaurora.tim.Constant;
import com.timaurora.tim.callback.TIMCb;

public class AuthHelper {
    private static String tag = AuthHelper.class.getSimpleName();
    public static void login(String identifier, String userSig, Promise promise){
        TIMUser user = new TIMUser();
        user.setIdentifier(identifier);
        //        user.setAccountType(Constant.ACCOUNT_TYPE+"");
        TIMManager.getInstance().login(Constant.SDK_APPID,user,userSig,new TIMCb("login",promise));
    }
    public static void logout(Promise promise){
        TIMManager.getInstance().logout(new TIMCb("logout",promise));
    }
}

File: ios/RCTIM/Helper/AuthHelper.m
//
// Created by 美德远健 on 2018/7/23.
// Copyright (c) 2018 Facebook. All rights reserved.
//

#import "AuthHelper.h"


@implementation AuthHelper {

}
+ (void)login:(NSString *)identifier userSig:(NSString *)userSig resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject {
    TIMLoginParam *param = [[TIMLoginParam alloc] init];
    [param setIdentifier:identifier];
    [param setUserSig:userSig];
    [param setAccountType:sdkAccountType];
    [param setAppidAt3rd:sdkAppId];
    [[TIMManager sharedInstance] login:param cb:[[TIMCb alloc] initWithName:@"login" resolve:resolve reject:reject]];
}

+ (void)logout:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject {
    TIMCb *cb = [[TIMCb alloc] initWithName:@"logout" resolve:resolve reject:reject];
    [[TIMManager sharedInstance] logout:^() {
        [cb onSucc];
    } fail:^(int code, NSString * err) {
        [cb onErr:code errMsg:err];
    }];
}


@end